integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -188,6 +210,55 @@ def _create_index(self, index_name: str) -> None:
         if self._index_client:
             self._index_client.create_index(index)
 
+    @staticmethod
+    def _serialize_metadata_fields(metadata_fields: Optional[Dict[str, type]]) -> Optional[Dict[str, str]]:
+        """Convert type objects to their string representations."""
+        if not metadata_fields:
+            return None
+        return {key: value.__name__ for key, value in metadata_fields.items()}
+
+    @classmethod
+    def _deserialize_metadata_fields(cls, fields: Optional[Dict[str, str]]) -> Optional[Dict[str, type]]:
+        """Convert string representations back to type objects."""
+        if not fields:
+            return None
+        try:
+            # Use the class-level TYPE_MAP for conversion.
+            ans = {key: cls.TYPE_MAP[value] for key, value in fields.items()}
+            return ans
+        except KeyError as e:
+            msg = f"Unsupported type encountered in metadata_fields: {e}"
+            raise ValueError(msg) from e
+
+    @staticmethod
+    def _serialize_index_creation_kwargs(index_creation_kwargs: Dict[str, Any]) -> Dict[str, Any]:
+        result = {}
+        for key, value in index_creation_kwargs.items():
+            if hasattr(value, "as_dict"):
+                result[key] = value.as_dict()
+            elif isinstance(value, list) and all(hasattr(item, "as_dict") for item in value):
+                result[key] = [item.as_dict() for item in value]
+            else:
+                result[key] = value
+        return result
```

Comment:
I don't know much about the details behind this - but either a value as an `as_dict` otherwise is kept as is - can we not bump into the case that we are not serializing something that should be?